### DISCUSSION TITLE
Get all intents available via the CLI

### DISCUSSION THREAD

**Discussion by goldyfruit** [OpenVoiceOS/ovos-buildroot#162] [Ideas]
One cool thing _(I don't know if it's possible)_, would be to have a list of all the intents available based on the installed skills, like `ovos-config get all-intents` _(or what ever sub-command :p)_.

**Comment by builderjer** [comment]
That shouldn't be hard.  Watching the logs go, you can see when each intent is registered and what it is.  I think it even breaks down the one-liner padatious intents

What would you use this for?  There is already suggestions that get displayed, although they only work with git clone skills right now.

**Reply by ChanceNCounter** [reply]
If it were pipeable, and therefore diffable and greppable, it’d be a huge help when dealing with a very large number of intents.

**Reply by denics** [reply]
When you use OVOS in another language you have to go and look into the translations to find the intents. getting the list in a certain language would be really helpful

**Comment by goldyfruit** [comment]
One use case will be to get the list of intents available (for one skill or all) from an external source to display it. This could be handle by a message on the bus with a filter for example.

**Comment by JarbasAl** [comment]
this is already in there, utils here https://github.com/OpenVoiceOS/ovos-utils/blob/dev/ovos_utils/intents/intent_service_interface.py#L323

**Reply by JarbasAl** [reply]
can add a new entrypoint script in ovos_utils setup.py listing this if desired, PR welcome!
